### DISCUSSION TITLE
How should we capitalize acronyms in camelCase and PascalCase?

### DISCUSSION THREAD

**Discussion by seancolsen** [mathesar-foundation/mathesar#908] [Developer Q&A] [answered]
I like that we have these [naming conventions](https://github.com/centerofci/mathesar/blob/42df7aa01310c0ff62b62165117caca54b7744b5/mathesar_ui/README.md#naming-conventions) (even though, as-written, they only apply to the front end code).

I'm wondering if we'd like to codify a decision pertaining to the capitalization of acronyms in camelCase and PascalCase.

I've noticed identifiers like `DbType`, `appendUrlPrefix` (which don't capitalize acronyms) and also identifiers like  `currentDBName`, and `QualifiedURIFunction` (which do capitalize acronyms).

The back end seems to use capitalized acronyms almost exclusively. The front end is more mixed. Since I'm mostly writing front-end code, I'm personally most interested in a decision that applies to the front end. We could decide on a rule for the codebase as a whole, though it doesn't necessarily need to have the same logic for front end and back end code. For example, we're already using snake_case variable on the back end and camelCase variables on the front end.

**Comment by seancolsen** [comment]
Personally, I have a medium-strong preference towards _keeping acronyms in lowercase_ when in camelCase and PascalCase.

That means I prefer names like `sendXhrRequest`, `MathesarJsonEncoder`, `syncActiveTabToUrl`, and `getApi`.

Why?

- I find lower case acronyms more readable.
- Adjacent acronyms don't work well when capitalized `getAPIURL` vs `getApiUrl`.
- At least in the JavaScript/TypeScript community, I find lowercase acronyms to be more common.
- Sometimes I use tooling in my editor the which converts between various casing. If I convert `raw_sql` from snake_case to camelCase, I get `rawSql` not `rawSQL`.
- Some words like `id` can end up on the fence. Is it an acronym, or an abbreviation, or just a word? Standardizing on lowercase seems more straightforward. (e.g. we have `getTabularTabByTabularID` and `dataFileId`)

**Comment by kgodey** [comment]
> We could decide on a rule for the codebase as a whole, though it doesn't necessarily need to have the same logic for front end and back end code. 

I think it makes sense for us to have different style guides for Python and JavaScript/TypeScript because those language communities have different conventions.

**Comment by pavish** [comment]
I have been mostly following capitalized acronyms. There are only a few places on the frontend where they are in lowercase for particular cases where using upper-cased acronyms cause readability issues.

Javascript/Typescript as a community does not have a standard convention for this. I've seen mostly mixed usage across projects. I do not have a strong preference towards either, but a slight inclination towards capitalized acronyms.

However, I find the arguments for always following lowercase acronyms as mentioned in https://github.com/centerofci/mathesar/discussions/908#discussioncomment-1849008 convincing enough and I think we can follow lowercase acronyms everywhere.

**Comment by seancolsen** [comment, marked as answer]
Based on the comments here, I'm preliminarily marking this as the answer:

| Context | Acronyms | Example |
| -- | -- | -- |
| Back end / Python | Upper case | `SQLQueryBuilder` |
| Front end / TypeScript | Lower case | `SqlQueryBuilder` |

We can re-evaluate if needed. 

I also don't think this means we need to go around changing all our code that deviates from this standard right away. I see enforcement of this standard as a very low priority. We can just try to keep it in-mind when writing new code and refactoring older code.

**Reply by kgodey** [reply]
Sounds good!
